jubilant/_pretty.py

```diff
@@ -0,0 +1,58 @@
+import dataclasses
+
+_MAX_VALUE = 150
+_INDENT = 2
+
+
+def _dump(value: object, indent='') -> str:
+    """Pretty-print a value with special cases for dataclasses, lists, and dicts.
+
+    If the value fits on a single (long) line, use that. Otherwise, split onto
+    multiple lines and indent each line.
+
+    For dataclasses, omit fields that are set to their default (or
+    default_factory return value).
+    """
```

Comment:
Are nested values allowed on the same line, like `{"a": {"b": {"c": 42}}}` ?

If so, then the function should somehow receive the amount of space already consumed to be used instead of indent if this is a sub-value.

Or, at least I can't see how this would work otherwise.

Example:
```
FirstLevel:
  SubLevelOne: a=23, b=45
  SubLevelTwo: a=88, b=99
```

When SubLevelTwo is rendered, and we're trying to see if the entire FirstLevel+subs would fit on a single line, that really depends on SubLevelOne length.

For another way to approach this, see the comment below.

Reply:
Ah yes, I think you're right there are some issues here. I'll take a closer look.

Reply:
I've taken a slightly different approach now, and added a comment in the code to clarify the part I think you're concerned about. Here's the docstring:

> If *value*'s fields or sub-values are simple scalar types and the result fits in a limited number of characters, use a single-line format. Otherwise, split onto multiple lines and indent each line.